internal/graph/check.go

```diff
@@ -805,64 +838,152 @@ func buildTupleKeyConditionFilter(ctx context.Context, reqCtx *structpb.Struct,
 // Finally, find the intersection between the two.
 // To use the fast path, we will need to ensure that the userset and all the children associated with the userset are
 // exclusively directly assignable. In our case, group member must be directly exclusively assignable.
-func (c *LocalChecker) checkUsersetFastPath(ctx context.Context, iter *storage.ConditionsFilteredTupleKeyIterator, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
+func (c *LocalChecker) checkUsersetFastPath(ctx context.Context, req *ResolveCheckRequest, iter *storage.ConditionsFilteredTupleKeyIterator) (*ResolveCheckResponse, error) {
 	ctx, span := tracer.Start(ctx, "checkUsersetFastPath")
 	defer span.End()
-
+	// Caller already verified typesys
 	typesys, _ := typesystem.TypesystemFromContext(ctx)
-	// We had just checked for the existence of the typesys in the caller.
-	// So, we are guaranteed for the presence of the context.
+	usersetDetails := buildUsersetDetails(typesys, "", tuple.GetType(req.GetTupleKey().GetUser()))
+	return c.checkMembership(ctx, req, iter, usersetDetails)
+}
 
-	reqUserType := tuple.GetType(req.GetTupleKey().GetUser())
+type usersets struct {
```

Comment:
can we rename this as `usersetsChanType` (or something like that) as usersets has various meaning